[Feat] MyPage UI

ssmj28 opened the pull request:
my page ui와 간단 기능 구현

**Check List :memo:**
- [x] Pull Request 제목 : [유형] 변경 사항 요약
- [x] 설명 & 이슈 번호 작성

**설명 (필수)**

이 PR에서 어떤 변경을 했는지 상세히 설명해주세요.
관련된 Issue 번호를 포함하세요.

1. #6 

2. 변경 사항
- 'mypage' 화면 UI 구현 [상단 타이틀, 친구 요청 목록, 프로필, 친구 목록]
- 간단한 기능 구현 [프로필 수정, 친구초대 링크 복사, 친구 추가 등] 

<br>

---

**review 참고 사항 (선택)**

추가로 알릴 내용이 있다면 여기에 작성해주세요.

1. 회원탈퇴 및 로그아웃 x

moon0727 committed:
feat: nickname textfield

- TextFields 파일 생성
- NickNameTextField 구현(특수문자, 숫자, 띄어쓰기 불가 + 8글자제한)

moon0727 committed:
feat: BirthdateField 구현

- 숫자8자리(띄어쓰기, 문자, 특수문자 불가)
- YYYY/MM/DD 형식 포맷팅 함수 DateTransformation 구현

moon0727 committed:
fix: 띄어쓰기 추가

- BirthdateField placeholder 띄어쓰기 추가

ssmj28 committed:
feat : mypage ui

my page ui와 간단 기능 구현

ssmj28 committed:
feat : mypage ui

my page ui와 간단 기능 구현

ssmj28 committed:
Update MypageScreen.kt

ssmj28 committed:
Update MypageScreen.kt

moon0727 committed:
feat : TravelTitleField 구현

- value, onValueChange, 여행상태, 여행상태변경함수를 받아 UI 조정

moon0727 committed:
feat: TravelDateField 구현

- YYYY/MM/DD 포맷팅 적용

ssmj28 committed:
Update MypageScreen.kt

ssmj28 committed:
Merge branch '9-feat-custom-textfield' into 6-feat-mypage-ui

ssmj28 committed:
feat : 친구추가팝업/마무리

전체적인 기능 수정 및 ui 마무리 했습니다.

ssmj28 committed:
Merge branch '1-feat-mypage-ui' of https://github.com/GolmokStar/frontend into 6-feat-mypage-ui

null4848 commented on app/src/main/java/com/example/golmokstar/ui/screens/MypageScreen.kt:
```diff
@@ -1,10 +1,906 @@
+                                                    }
+                                                },
+                                                modifier = Modifier
+                                                    .size(50.dp, 20.dp)
```
여행 스타일 선택할 때랑 여행 스타일 표시할 때랑 크기가 다른데 맞추는 게 어떨까요 
스타일 표시할 때는 65 25 라운드코너도 25 입니다!

null4848 commented on app/src/main/java/com/example/golmokstar/ui/screens/MypageScreen.kt:
```diff
@@ -1,10 +1,906 @@
+                                                    )
+                                                    .padding(0.dp),
+                                                colors = ButtonDefaults.buttonColors(
+                                                    if (selectedStyles.contains(style)) Color.White else MainNavy,
```
선택할 때 색이 네이비가 되는 게 좋을 것 같아요! 
지금은 살짝 헷갈립니다!

null4848 commented on app/src/main/java/com/example/golmokstar/ui/screens/MypageScreen.kt:
```diff
@@ -1,10 +1,906 @@
+                        }
+                    }
+                }
+            }
```
Circle 크기를 좀 더 키우시거나 Spacer도 더 주시면 좋을 것 같아요! 조금 더 넓게 퍼져 있는 느낌이면 더 좋을 것 같습니다

null4848 commented on app/src/main/java/com/example/golmokstar/ui/screens/MypageScreen.kt:
```diff
@@ -1,10 +1,906 @@
+                        label = {
+                            Text(
+                                text = "회원번호를 입력해주세요. ex) #0000",
+                                style = AppTypography.labelMedium,
```
피그마상에서 글씨 더 커요! bodyMedium으로 생각하시면 될 것 같습니다.

null4848 commented on app/src/main/java/com/example/golmokstar/ui/screens/MypageScreen.kt:
```diff
@@ -1,10 +1,906 @@
+
+        Spacer(modifier = Modifier.width(8.dp))
+
+        // Travel Styles
```
친구 목록에서 스타일을 표시할 때도 프로필에 표시하는 것처럼 labelMedium 글씨체로 맞추는 게 좋을 것 같아요

null4848 commented on app/src/main/java/com/example/golmokstar/ui/screens/MypageScreen.kt:
```diff
@@ -1,10 +1,906 @@
+                .clickable { onBellClick() }
+        )
+    }
+}
```
이 부분에도 top padding(10.dp) 넣어주세요!

null4848 reviewed (commented):
UI가 복잡하고 기능도 많은데 정말 고생 많으셨어요! 구현해주신 기능은 완벽한 것 같습니다. UI 디테일만 한 번 더 확인해서 수정해주시면 좋을 것 같고 추가로 친구 삭제 기능도 넣어주시면 될 것 같아요!

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/MypageScreen.kt:
```diff
@@ -1,10 +1,906 @@
+                textAlign = TextAlign.Center
+            )
+
+            Divider(
```
HorizontalDivider로 바꾸면 좋을 거 같아요~

moon0727 commented on app/src/main/java/com/example/golmokstar/ui/screens/MypageScreen.kt:
```diff
@@ -1,10 +1,906 @@
+                        shape = RectangleShape,
+                        onClick = {
+                            if (membershipNumber.length < 5) { // # 포함 최소 5자리 (#0000)
+                                errorMessage = "* 회원 번호를 입력해 주세요."
```
* 존재하지 않는 회원번호입니다.
도 추가 부탁드립니다!

ssmj28 committed:
feat : comment

- 수정 모드 여행 스타일 선택 레이아웃
- 수정 모드 프로필 사진 선택 레이아웃
- 친구 추가 팝업 레이아웃 및 안내 문구 추가
- 친구 목록 스타일 레이아웃
- 마이페이지 상단 패딩
- 친구 요청 팝업 레이아웃
- 친구 삭제
